src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -504,7 +503,7 @@ private boolean hasEmptyLineAfter(DetailAST token) {
             lastToken = token.getLastChild();
         }
         DetailAST nextToken = token.getNextSibling();
-        if (isComment(nextToken)) {
+        if (lastToken != null && isComment(nextToken)) {
```

Comment:
You added checking if lastToken is not null but few lines after this lastToken is used to calculate currentEnd:
```java
          if (lastToken != null && isComment(nextToken)) {
                     final int currentEnd = lastToken.getLineNo(); 
                     nextToken = nextToken.getNextSibling();
          }
          // Start of the next token
         final int nextBegin = nextToken.getLineNo();
         // End of current token.
         final int currentEnd = lastToken.getLineNo();
         return hasEmptyLine(currentEnd + 1, nextBegin - 1);
     }
```
This does not make you suspicious that something is clearly wrong?